apps/studio/state/addons-page.ts

```diff
@@ -0,0 +1,25 @@
+import { useUrlState } from 'hooks/ui/useUrlState'
+
+export const ADDONS_PANEL_KEYS_ARRAY = ['computeInstance', 'pitr', 'customDomain', 'ipv4'] as const
+
+export type ADDONS_PANEL_KEYS = undefined | (typeof ADDONS_PANEL_KEYS_ARRAY)[number]
+
+export function useAddonsPagePanel() {
+  const [{ panel: _panel }, setParams] = useUrlState({ replace: true })
```

Comment:
Small nit: this may cause issues if there's something else in the query params. You need to extract the rest of the params and use them when calling `setParams`, right?

I couldn't find any issues at the moment in the pages you use this, though.

Reply:
`useUrlState()` is nice enough to handle this for us! So it'll lave any other query params untouched when calling `setParams`